### ISSUE TITLE
Metric tracker should track a secondary metric when the primary metric is equal between epochs

### ISSUE THREAD

**Issue by bratao** [allenai/allennlp#4524]
**Is your feature request related to a problem? Please describe.**
I want to train a model that maximize f-1, but when two epochs have the same f-1, check is loss decreased.

Some models that I'm training achieve a f-1 of 1.0. but there is still potential for a decrease in loss.

**Describe the solution you'd like**
That the metric tracker allows to use a secondary metric to track.

Would AllenNLP accept a PR with this feature or is out of scope?

**Comment by matt-gardner** [comment]
Yes, this is something we'd accept a PR for.  I would recommend changing the type of `metric_name` to `Union[str, List[str]]`.

**Comment by pab-vmware** [comment]
We were checking to see if we could implement this but,
@dirkgr you seem to have implemented it in https://github.com/allenai/allennlp/pull/4800/files. 
Please check.

**Comment by bratao** [comment]
For me, yes!
